src/main/java/com/myapp/impl/ProviderDaoImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.myapp.impl;
+
+import com.myapp.dao.ProviderDao;
+import com.myapp.jdbc.*;
+import com.myapp.model.Item;
+import com.myapp.model.Provider;
+import com.myapp.model.UpdateProvider;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class ProviderDaoImpl implements ProviderDao {
+	private static final Log logger = LogFactory.getLog(ProviderDaoImpl.class);
+
+	@InjectRandomInt(min = 2, max = 7)
+	private int repeat;
+
+	private String message;
+
+	public void setMessage(String message) {
+		this.message = message;
+	}
+
+	@Autowired
```

Comment:
Field injection is not recommended 
http://blog.marcnuri.com/field-injection-is-not-recommended/ 

read about constructor Injection in Spring with lombok
https://www.baeldung.com/spring-injection-lombok